modules/fakemeta/pdata.cpp

```diff
@@ -192,30 +192,35 @@ static cell AMX_NATIVE_CALL set_pdata_string(AMX *amx, cell *params)
 		szData = get_pdata_direct<char*>(pEdict, iOffset);
 		if (IsBadWritePtr(szData, 1))
 			return 0;
-		strcpy(szData, data);
 	} else {
 		szData = get_pdata<char*>(pEdict, iOffset);
 		if (IsBadWritePtr(szData, 1))
 			return 0;
-		if (params[4] == 1)
+
+		if (len > static_cast<int>(strlen(szData)))
 		{
-			free(szData);
-			szData = (char *)malloc(len + 1);
-		} else if (params[4] == 2) {
-			delete [] szData;
-			szData = new char[len + 1];
+			if (params[4] == 1)
+			{
+				free(szData);
+				szData = (char *)malloc(len + 1);
+			}
+			else if (params[4] == 2) {
+				delete[] szData;
+				szData = new char[len + 1];
```

Comment:
Not related to this PR, but who will `free`/`delete[]` this data? Should we have some storage for allocated strings? (and free them on map change?)

Reply:
I think the library should care for it automatically, no?
I don't mind a storage, but not sure if it makes actually a difference.

Reply:
@Arkshine do you mean gamedll? I'm worry a bit about of compatibility between gamedll's `delete[]` and amxmodx's `delete[]`, but I hope they use the same since there is the `libstdc++` located in the server's root directory.